scripts/generate_gallery.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python3
+"""
+Sphinx plugin to generate a gallery for notebooks
+
+Modified from the pytensor project, which was modified from the pymc project,
+which modified the seaborn project, which modified the mpld3 project.
+"""
+
+import base64
+import json
+import os
+import shutil
+import tempfile
+from pathlib import Path
+
+try:
+    import matplotlib
+
+    matplotlib.use("Agg")
+    import matplotlib.pyplot as plt
+    from matplotlib import image
+
+    MATPLOTLIB_AVAILABLE = True
+except ImportError:
+    print("WARNING: Matplotlib not available. Using fallback for gallery generation.")
+    MATPLOTLIB_AVAILABLE = False
+
+# Define directories
+PROJECT_ROOT = Path(__file__).resolve().parent.parent
+NOTEBOOK_DIR = PROJECT_ROOT / "docs" / "source" / "notebooks"
+GALLERY_DIR = PROJECT_ROOT / "docs" / "source" / "gallery"
+GALLERY_IMG_DIR = GALLERY_DIR / "images"
+
+# Create gallery images directory if it doesn't exist
+GALLERY_IMG_DIR.mkdir(exist_ok=True, parents=True)
+
+# Default image in case we can't extract one from a notebook
+DEFAULT_IMG_LOC = PROJECT_ROOT / "docs" / "source" / "_static" / "flat_logo.png"
+
+# Map folder names to display titles
+folder_title_map = {
```

Comment:
Is the secretly used? I don't see a renference

Reply:
good catch! removed in https://github.com/pymc-labs/pymc-marketing/pull/1611/commits/f24602e59ee43f64eaabc0787a59cacc58b5400b